airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -134,12 +139,32 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
         return;
       }
 
-      // if the buffer flushes, update the states appropriately.
-      // TODO: ryankfu (if true, this is where bundling up logic to also commit to airbyte_raw table)
-      if (bufferingStrategy.addRecord(stream, message)) {
-        markStatesAsFlushedToTmpDestination();
+      final Optional<BufferFlushType> flushType = bufferingStrategy.addRecord(stream, message);
+      // if present means that a flush occurred
+      if (flushType.isPresent()) {
+        if (flushType.equals(BufferFlushType.FLUSH_ALL)) {
+          // when all buffers have been flushed then we can update all states as flushed
+          markStatesAsFlushedToDestination();
+        } else if (flushType.equals(BufferFlushType.FLUSH_SINGLE_STREAM)) {
+          if (stateManager instanceof DestStreamStateLifecycleManager) {
```

Comment:
mild preference to add some sort of "supports per-stream flush" method to state lifecycle manager, rather than explicity doing an instanceof check